### PR TITLE
Feature/filestorage availabilityzones list

### PR DESCRIPTION
**Pull request by ehdou**
Add support for Availability Zones List
For #152

### CODE REVIEW

**Comment by ehdou** [comment]
[Availability zones](https://github.com/openstack/manila/blob/16c522bf5e5c3d37018caef11dd7852462193327/manila/api/v2/availability_zones.py) in Manila code

**Comment by ehdou** [comment]
@jrperritt This PR is ready for review. It adds support for availability zones List

File: openstack/sharedfilesystems/v2/availabilityzones/results.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	// The date and time stamp when the availability zone was updated.
+	UpdatedAt gophercloud.JSONRFC3339MilliNoZ `json:"updated_at"`
+}
+
+type commonResult struct {
+	gophercloud.Result
+}
+
+// ListResult contains the response body and error from a List request.
+type ListResult struct {
```
Comment (jrperritt) [inline openstack/sharedfilesystems/v2/availabilityzones/results.go]: Should be `AvailabilityZonePage`.
Should be a page type.

File: openstack/sharedfilesystems/v2/availabilityzones/results.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+// AvailabilityZone contains all the information associated with an OpenStack
+// AvailabilityZone.
+type AvailabilityZone struct {
+	// The availability zone ID.
+	ID string `json:"id"`
+	// The name of the availability zone.
+	Name string `json:"name"`
+	// The date and time stamp when the availability zone was created.
+	CreatedAt gophercloud.JSONRFC3339MilliNoZ `json:"created_at"`
```
Comment (jrperritt) [inline openstack/sharedfilesystems/v2/availabilityzones/results.go]: We're making all times of type `time.Time`. See #190 for how to handle these.

File: openstack/sharedfilesystems/v2/availabilityzones/requests.go [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package availabilityzones
+
+import "github.com/gophercloud/gophercloud"
+
+// List will return the existing availability zones.
+func List(client *gophercloud.ServiceClient) (r ListResult) {
```
Comment (jrperritt) [inline openstack/sharedfilesystems/v2/availabilityzones/requests.go]: `List` should return an unnamed `pagination.Pager`.

**Comment by ehdou** [comment]
@jrperritt this PR has been rebased, comments have been addressed. It's now ready for review

**Pull request merged**